File: Python3/Alternativas/ej12.py
# -*- coding: utf-8 -*-

#Escribir un programa que lea un año indicar si es bisiesto. 
#Nota: un año es bisiesto si es un número divisible por 4, pero no si 
#es divisible por 100, excepto que también sea divisible por 400.

#1.almacenar año (entero)
#2.comprobar que es divisible por 4
#3.comprobar si es divisible por 100
#4.si lo es no es bisiesto
#5.comprobar si es divisible por 400
#6.si lo es, es bisiesto

print("--->Comprobar si un año es bisiesto<---")
anno = int(input("Introduce el año> "))
bisiesto = False
if(anno%4==0):
    if(anno%100==0):
        bisiesto = False
        if(anno%400==0):
            bisiesto = True
if(bisiesto==True):
    print(f"El año {anno} es bisiesto")
else:
	print(f"El año {anno} no es bisiesto")

File: Java/Alternativas/ej12.java
import java.util.Scanner; //Importar la clase Scanner

/**
 * Escribir un programa que lea un año indicar si es bisiesto. 
 * Nota: un año es bisiesto si es un número divisible por 4, 
 * pero no si es divisible por 100, excepto que también sea divisible por 400.
 * @author Alberto Miguel Martinez Jimenez
 */

/**
 * Algoritmo
 * Almacenar año
 * es bisiesto si anno%4=0
 * excepto anno%100=0
 * pero no si se da anno%400=0 
 */
public class ej12 {

	public static void main(String[] args) {
		int anno;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("---Comprobar si un año es bisiesto---");
		System.out.println("Introduce un año(Ej.:1994)> ");
		anno = scanner.nextInt();
		if((anno%4==0)&&(anno%100!=0)) {
			System.out.println("El año "+anno+" es bisiesto");
			
		}else if((anno%400==400)){
			System.out.println("El año "+anno+" es bisiesto");
		}else{
			System.out.println("El año "+anno+" no es bisiesto");
		}
	}

}
